internal/runner/service.go

```diff
@@ -216,7 +218,20 @@ func (r *runnerService) Execute(srv runnerv1.RunnerService_ExecuteServer) error
 		return errors.WithStack(err)
 	}
 
-	logger.Debug("received initial request", zap.Any("req", req))
+	reqJSON, err := protojson.Marshal(req)
+	if err != nil {
+		logger.Error("failed to marshal request", zap.Error(err))
+	}
+
+	// We want to always log the request because it is used for AI training.
```

Comment:
We should try to add a tests for his in https://github.com/jlewi/runme/blob/jlewi/logs/internal/runner/service_test.go - the reason being is that I'm currently in the middle of charting a migration from `runnerv1` (this impl) to `runnerv2` (alpha; future) and the tests serve as a spec to not have things fall thru the cracks.

`runnerv2` lives over here https://github.com/jlewi/runme/tree/jlewi/logs/internal/runnerv2service but instead of syncing the service impl, I'd rather have a comprehensive test suite in `runnerv1` as guard rails for filling in all the gaps.

I'm happy to take a stab at adding a test case if you need a template.

Reply:
Let me take a look at this.

Reply:
@sourishkrout added a test. The test passes locally when I run it individually and the tests appear to be green in CI. However, as noted in #587 I'm seeing test failures when running locally

Reply:
Apologies for the #587 expedition. After rerunning the C/I job it looks like it's indeed the flaking issue we are working on.